lib/misc/firestore.dart

```diff
@@ -8,6 +13,37 @@ class Firestore {
 
   final user = Auth.instance.currentUser;
 
+  Future<int> getStreak() async {
+    final docSnapshot = await FirebaseFirestore.instance
+        .collection('users')
+        .doc(user!.uid)
+        .get();
+    return docSnapshot.data()?['streak'];
+  }
+
+  Future<void> checkStreak() async {
+    if (user == null) {
+      return;
+    }
+    final docSnapshot = await FirebaseFirestore.instance
+        .collection('users')
+        .doc(user!.uid)
+        .get();
+    final temp = docSnapshot.data()?['last_challenge_completed'];
```

Comment:
tady zase, celkově si na Firebase dávej v tomhle ohledu pozor. Ono se to tváří strašně jednoduché a fajn, ale když dojde na typovost, dokáže dost potrápit. Nejlepší lék na tohle je si ty data deserializovat do nějakých modelů, který budou mít nějaké default hodnoty nebo ošetření pro null hodnoty (pro nějaké budoucí projekty 😉 ).